pkg/metrics/metrics.go

```diff
@@ -79,6 +114,38 @@ func (mm *metricsManager) EmitGKEComponentVersion() error {
 	return nil
 }
 
+// CodeForError returns a pointer to the grpc error code that maps to the http
+// error code for the passed in user googleapi error. Returns codes.Internal if
+// the given error is not a googleapi error caused by the user. The following
+// http error codes are considered user errors:
+// (1) http 400 Bad Request, returns grpc InvalidArgument,
+// (2) http 403 Forbidden, returns grpc PermissionDenied,
+// (3) http 404 Not Found, returns grpc NotFound
+// (4) http 429 Too Many Requests, returns grpc ResourceExhausted
+func CodeForError(err error) *codes.Code {
```

Comment:
Why are we copying this code over? We can use the existing CodeForError in the pkg/gce-pd-csi-driver/utils.go?

Reply:
The CodeForError in the pkg/gce-pd-csi-driver/utils.go resulted in circular dependency thus moving it to pkg/common/utils.go as discussed